### PR TITLE
Weather

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Primarily tg code but tweaking it to be better

## Why It's Good For The Game

Immursiun

## Changelog
:cl:
add: Weather
add: Analyzers can be used inhand to predict weather
add: Ice Colony and Icy caves now have snow
add: Big red now has not-darude sandstorms; remember to get a mask to protect yourself from the stronger storms!
add: LV has rain
code: Removed a useless area var
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
GIV

File: _maps/ice_colony_v2.json [hunk 1]
```diff
@@ -6,5 +6,8 @@
         "COLD": true
     },
     "armor": "ice",
-    "announce_text": "A garbled, unintelligible communications message was broadcasted over a general frequency, and picked up by our comms relay. The message appears to have come from a second generation settlement, located on an ice cold planet. The ###SHIPNAME### is moving into the sector with thrusters at max throttle. TGMC, get briefed and then move out!"
+	"announce_text": "A garbled, unintelligible communications message was broadcasted over a general frequency, and picked up by our comms relay. The message appears to have come from a second generation settlement, located on an ice cold planet. The ###SHIPNAME### is moving into the sector with thrusters at max throttle. TGMC, get briefed and then move out!",
+	"traits":[{
+		"Weather_Snowstorm": true
```
Comment (psykzz) [inline _maps/ice_colony_v2.json:11]: lowercase this, it will just help in the long run.

File: code/datums/weather/weather_types/sand_storm.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	weather_message = "<span class='notice'>Gentle sand wafts down around you like grotesque snow. The storm seems to have passed you by...</span>"
+	weather_overlay = "dust_med"
+
+	end_message = "<span class='notice'>The sandfall slows, stops. Another layer of sand on the mesa beneath your feet.</span>"
+	end_overlay = "dust_low"
+
+	aesthetic = TRUE
+
+	probability = 70
```
Comment (psykzz) [inline code/datums/weather/weather_types/sand_storm.dm:41]: This didn't have a telegraph

**Comment by psykzz** [comment]
Weather is covered by weeds

File: code/datums/weather/weather_types/sand_storm.dm (hunk 2 above)
Comment (psykzz) [reply code/datums/weather/weather_types/sand_storm.dm:41]: Or at least didn't feel like it did

File: code/datums/weather/weather_types/acid_rain.dm [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+/datum/weather/acid_rain/telegraph()
+	. = ..()
+	var/list/eligible_areas = list()
+	for (var/z in impacted_z_levels)
+		eligible_areas += SSmapping.areas_in_z["[z]"]
+
+	midsound.output_atoms = eligible_areas
+
+/datum/weather/acid_rain/start()
+	midsound.start()
```
Comment (psykzz) [inline code/datums/weather/weather_types/acid_rain.dm:40]:
```suggestion
	. = ..()
	midsound.start()
```

**Comment by XSlayer300** [comment]
Hmm, any rain for WO? How about a fog weather?

**Comment by psykzz** [comment]
This has highlighted an issue with the mapping loader.
When you provide traits it doesn't set the default GROUND trait (but adds station instead).

This causes the problems with distress, and new larva since levels were not considered "gameplay" areas

**Comment by TiviPlus** [comment]
TM/ Merge only with https://github.com/tgstation/TerraGov-Marine-Corps/pull/4664

**Comment by jroinc** [comment]
Acid rain killed everyone in the FOB on LV before shutters went up. In like 15 seconds.

Maybe roof the LZ's/prime FOB spots and nerf the weather damage a bit? Either that or clearer signals that it's gonna be "acid" rain as opposed to just normal rain. We coulda gone onboard the Alamo if it had been clear that this was lethal.

**Comment by psykzz** [comment]
Another option would be to increase damage over time of the efffect.
So even if it is acid rain and you missed the announcement you have a ramping efffect that gives you time to get sheltered

**Review by TheBimmer231 — APPROVED** [review]
Usable, not bad.

**Review by psykzz — APPROVED** [review]

**Pull request merged**